### PR TITLE
Improve the way we get instances with a tag from helix

### PR DESCRIPTION
**Pull request by npawar**
Get all instance configs in one shot, and then filter out the instance configs based on tags/enabled. This will avoid exceptions we encounter if an instances is in an inconsistent state (for example present in INSTANCES but not in PARTICIPANTS). This is also more efficient than fetching all instances from INSTANCES first and then fetching each instance config

### CODE REVIEW

File: pinot-common/src/main/java/com/linkedin/pinot/common/utils/helix/HelixHelper.java [hunk 1]
```diff
@@ -410,4 +410,22 @@ public IdealState apply(IdealState idealState) {
     }
     return enabledInstances;
   }
+
+  public static List<String> getInstancesWithTag(HelixAdmin helixAdmin, String helixClusterName,
```
Comment (kishoreg) [inline pinot-common/src/main/java/com/linkedin/pinot/common/utils/helix/HelixHelper.java]: lets get all instanceConfigs in one shot. that should not throw any exceptions and is more efficient.

Comment (npawar) [reply pinot-common/src/main/java/com/linkedin/pinot/common/utils/helix/HelixHelper.java]: I don't think I understand. That's exactly what this method is doing

Comment (npawar) [reply pinot-common/src/main/java/com/linkedin/pinot/common/utils/helix/HelixHelper.java]: Understood what you're referring to from the conversation offline. Will change it accordingly

Comment (npawar) [reply pinot-common/src/main/java/com/linkedin/pinot/common/utils/helix/HelixHelper.java]: @kishoreg made the changes to get all instance configs in one shot

**Comment by npawar** [comment]
Moved code from both methods to a common private method

**Pull request merged**